packages/golden_toolkit/lib/src/testing_tools.dart

```diff
@@ -128,29 +128,37 @@ void testGoldens(
   });
 }
 
-/// This [screenMatchesGolden] is wrapper on top of [matchesGoldenFile]
+/// A function that wraps [matchesGoldenFile] with some extra functionality. The function finds the first widget
+/// in the tree if [finder] is not specified. Furthermore a [fileNameFactory] can be used in combination with a [name]
+/// to specify a custom path and name for the golden file. In addition to that the function makes sure all images are
+/// available before
 ///
-/// [goldenFileName] is a file name output, must NOT include extension like .png
+/// [name] is the name of the golden test and must NOT include extension like .png. Use [fileNameFactory] to construct
+/// an actual file path from that name. If not [fileNameFactory] is specified the [GoldenToolkitConfiguration]'s one is
+/// used, which defaults to `goldens/$name.png`.
+///
+///
+/// [finder] is an optional finder, which can be used to target a specific widget to use for the test. If not specified
+/// the first widget in the tree is used
 ///
 /// [autoHeight] tries to find the optimal height for the output surface. If there is a vertical scrollable this
 /// ensures the whole scrollable is shown. If the targeted render box is smaller then the current height, this will
 /// shrink down the output height to match the render boxes height.
 ///
-/// [finder] optional finder
-///
 /// [customPump] optional pump function, see [CustomPump] documentation
 ///
 /// [skip] by setting to true will skip the golden file assertion. This may be necessary if your development platform is not the same as your CI platform
 Future<void> screenMatchesGolden(
   WidgetTester tester,
-  String goldenFileName, {
+  String name, {
   bool autoHeight,
+  FileNameFactory fileNameFactory,
```

Comment:
similar to above... let's nuke the method level override and rely on the global configuration

Reply:
This would need a major refactoring because we then cannot pass in `configuration.deviceNameFactory` to `screenMatchesGolden`.

Reply:
Or we have to change the configuration for the duration of the method call

Reply:
if they both rely exclusively on the configuration singleton, then there shouldn't be an issue, right? Or, perhaps I am missing something.

Reply:
The problem is `multiScreenGolden` uses `screenMatchesGolden`. `screenMatchesGolden` has no knowledge about any device. But `multiScreenGolden` should call `deviceFileNameFactory` while `screenMatchesGolden` should call `fileNameFactory`

Reply:
Okay gotcha... I see... so I think there may be some simplification we can do.

What if we collapsed down to a single filename factory (the deviceFileNameFactory) but make it null tolerant. 

Then instead of having multiScreenGolden invoke the public screenMatchesGolden API... what if we pushed the implementation of screenMatchesGolden into a private shared func that took an optional device parameter?

then the public screenMatchesGolden function is left mostly as a pass through function.

I think it should be a pretty quick fix.

Reply:
> what if we pushed the implementation of screenMatchesGolden into a private shared func that took an optional device parameter?

Maybe this function should be public and already implement a constraints based approach. 
I will try to sort my thoughts and come up with something.

Reply:
Something like this would be really cool:

```dart
expect(tester.getGoldens(sizes: [phone, tablet]), matchGoldenFiles('screen1'))
```

I will try to figure out if we can do sth like this! The screenMatchesGolden and multiScreenGolden methods should then be implemented using this internally

Reply:
I'm inclined to keep this PR as targeted as possible and not widen the public API too far. 

We are planning to overhaul the public API and deprecate screenMatchesGolden and multiScreenGolden functions and replace them a singular API in the near future (for a 1.0 release). However, I don't think we are quite ready to do that. We've been waiting a little bit to try to understand how people are using the package first so that we make a more informed decision and create a more stable API. 

For now, if we achieve the code sharing via private methods, then we keep it an internal implementation detail of the package that is free to change in a backwards compatible way.

Reply:
> Something like this would be really cool:
> 
> ```dart
> expect(tester.getGoldens(sizes: [phone, tablet]), matchGoldenFiles('screen1'))
> ```
> 
> I will try to figure out if we can do sth like this! The screenMatchesGolden and multiScreenGolden methods should then be implemented using this internally

that would be neat in that it would be more composable.

Reply:
> I'm inclined to keep this PR as targeted as possible and not widen the public API too far.

You are totally right I'll just implement a private function for now.

> that would be neat in that it would be more composable.

And it looks more like the dart way of writing expectation tests.